src/servers/sapphire_zone/Actor/Player.cpp

```diff
@@ -1606,11 +1647,17 @@ void Core::Entity::Player::finishZoning()
    switch( getZoningType() )
    {
       case ZoneingType::None:
-         sendToInRangeSet( ActorControlPacket143( getId(), ZoneIn, 0x01 ), true );
+         if (getGmInvis() == false)
```

Comment:
better way to do this would be to have a check before the switch statement and check displayflags for invisible and then just return if its set, although this has some issues too

might need to rewrite this a bit so GMs can always see GMs without having to duplicate checks everywhere like you've done here :p

Reply:
ah I kinda half wrote what I was thinking and left out the juicy stuff

what I was getting at is that duplicating all this is pretty bad, would be better to rewrite this so the packet code isn't duplicated 4 times along with the possibility for GMs to see invis GMs too

Reply:
Like it has already been stated, i am very hesitant to merge this as it should be implemented using the priorities.